punish/punish.py

```diff
@@ -1166,6 +1227,24 @@ def execute_queue_event(self, *args) -> bool:
         until = (now + duration + 0.5) if duration else None
         duration_ok = (case_min_length is not None) and ((duration is None) or duration >= case_min_length)
 
+        if member.id not in self.json[server.id]:
+            # remove all roles from user that are specified in remove_role_list, only if its a new punish
+            user_roles = member.roles
+            removed_user_roles = []
+            removed_parsed_user_roles = []
+
+            for remove_role in remove_role_list:
+                parsed_removed_role = self._role_from_string(server, remove_role)
+                if parsed_removed_role in user_roles:
+                    removed_user_roles.append(remove_role)
+                    removed_parsed_user_roles.append(parsed_removed_role)
+                elif parsed_removed_role is None:
+                    await self.bot.say("The {} role was not found, please update the remove role list!".format(remove_role))
```

Comment:
If the role was deleted, it's better to silently ignore it, both when listing the roles and when re-adding them. This is much less error-prone if IDs are used instead of names, of course.